File: Data Structures & Algorithms/majority-element/submission-0.java
class Solution {
    public int majorityElement(int[] nums) {
        int res=0, max=0;
        HashMap<Integer,Integer> seen = new HashMap<>();
        for (int i = 0;i < nums.length;i++){
            seen.put(nums[i], seen.getOrDefault(nums[i], 0) + 1);
            if (seen.get(nums[i]) > max){
                res = nums[i];
                max = seen.get(nums[i]);
            }
            }
        return res;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        res = [0] * n
        if n <= 1:
            return n
        for i in range(n):
            prod = 1
            for j in range(n):
                if i == j:
                    continue
                prod *= nums[j]
            res[i] = prod
        return res        